apps/etterlatte-saksbehandling-ui/client/src/components/tilbakekreving/vurdering/TilbakekrevingVurderingSkjema.tsx

```diff
@@ -58,33 +56,83 @@ export function TilbakekrevingVurderingSkjema({
   behandling: TilbakekrevingBehandling
   redigerbar: boolean
 }) {
-  const dispatch = useAppDispatch()
+  if (!behandling) {
+    return
+  }
   const [lagreVurderingStatus, lagreVurderingRequest] = useApiCall(lagreTilbakekrevingsvurdering)
-  const { register, handleSubmit, watch, control, getValues, setValue } = useForm<TilbakekrevingVurdering>({
-    defaultValues: behandling.tilbakekreving.vurdering ? behandling.tilbakekreving.vurdering : initialVurdering,
-    shouldUnregister: true,
-  })
+
+  const { register, handleSubmit, watch, control, getValues, setValue, formState, reset } =
+    useForm<TilbakekrevingVurdering>({
+      defaultValues: behandling.tilbakekreving.vurdering || initialVurdering,
+    })
 
   useEffect(() => {
-    if (watch().vilkaarsresultat === TilbakekrevingVilkaar.IKKE_OPPFYLT) {
-      setValue('hjemmel', TilbakekrevingHjemmel.TJUETO_FEMTEN_FEMTE_LEDD)
-    } else {
-      setValue('hjemmel', watch().rettsligGrunnlag)
+    const values = getValues()
+    const utledetHjemmel =
+      values.vilkaarsresultat === TilbakekrevingVilkaar.IKKE_OPPFYLT
+        ? TilbakekrevingHjemmel.TJUETO_FEMTEN_FEMTE_LEDD
+        : values.rettsligGrunnlag
+    if (values.hjemmel !== utledetHjemmel) {
+      setValue('hjemmel', utledetHjemmel, { shouldDirty: false })
     }
-  }, [watch().vilkaarsresultat, watch().rettsligGrunnlag])
 
-  const lagreVurdering = (vurdering: TilbakekrevingVurdering) => {
-    lagreVurderingRequest({ behandlingsId: behandling.id, vurdering }, (lagretTilbakekreving) => {
-      dispatch(addTilbakekreving(lagretTilbakekreving))
-    })
+    setIkkeRelevanteFelterTilNull(
+      values,
+      setValue,
+      skalHaForhaandsvarsel(),
+      rettsligGrunnlagForVilkaarOppfyltEllerDelvisOppfylt(),
+      vilkaarOppfylt(),
+      beloepIBehold()
+    )
+
+    if (formState.isDirty && Object.keys(formState.dirtyFields).length) {
+      const delay = setTimeout(() => {
+        lagreVurdering({ vurdering: values, automatisk: true })
+      }, 3000)
+
+      return () => clearTimeout(delay)
+    }
+  }, [watch()])
```

Comment:
Flere ting her: 
- Hvorfor depender du på watch og ikke formState som du faktisk bruker her? 
- Må ikke depende med `()`
- Hvis du depender på watch må du også sørge for å unsubscribe til den for å ikke få minneproblemer. 

Mer i dokumentasjonen: https://react-hook-form.com/docs/useform/watch

Reply:
Takk. Jeg trenger visst ikke å depende på watch nå :) Gått tilbake til formstate i ny commit